torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -938,21 +938,7 @@ std::shared_ptr<JitFuture> TensorPipeAgent::send(
                          << clientPipe.pipe_->getRemoteName() << ": "
                          << error.what();
           }
-          std::shared_ptr<AtomicJitFuture> futureResponseMessage;
-          {
-            std::lock_guard<std::mutex> lock(clientPipe.mutex_);
-            auto pendingFutIt =
-                clientPipe.pendingResponseMessage_.find(messageId);
-            TORCH_INTERNAL_ASSERT(
-                pendingFutIt != clientPipe.pendingResponseMessage_.end(),
-                "message ID ",
-                messageId,
-                " is not recognized");
-            futureResponseMessage = std::move(pendingFutIt->second);
-            clientPipe.pendingResponseMessage_.erase(pendingFutIt);
-          }
-          markFutureWithError(futureResponseMessage, error.what());
-          removeFromTimeoutMap(messageId);
+          handleClientError(clientPipe, error);
```

Comment:
~Would I be correct if I assume that, after doing this, the pipe is still alive and it can still send/recv future messages? And the impact is only on pending ones?~ doesn't look so